packages/checkout/widgets-lib/src/widgets/checkout/hooks/useRouteUpdatedRelayer.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useRef } from 'react';
+import { useCheckoutContext } from '../context/CheckoutContextProvider';
+import { CheckoutActions } from '../context/CheckoutContext';
+
+export const useRouteUpdatedRelayer = () => {
+  const [{ postMessageHandler }, checkoutDispatch] = useCheckoutContext();
+  const unsubscribePostMessageHandler = useRef<() => void>();
+
+  useEffect(() => {
+    if (!postMessageHandler) return;
+    unsubscribePostMessageHandler.current?.();
+    unsubscribePostMessageHandler.current = postMessageHandler?.subscribe(({ type, payload }) => {
+      if (type !== 'IMTBL_ROUTE_UPDATED' as any) return;
```

Comment:
todo: add to types once https://github.com/immutable/ts-immutable-sdk/pull/2111 is merged to avoid conflicts